packages/solana/src/connectors/WalletConnectConnector.ts

```diff
@@ -0,0 +1,254 @@
+  currency: 'SOL',
+  explorerUrl: 'https://solscan.io',
+  rpcUrl: 'https://api.testnet.solana.com'
+}
+
+export const solanaDevnet = {
+  chainId: '8E9rvCKLFQia2Y35HXjjpWzj8weVo44K',
+  name: 'Solana Devnet',
+  currency: 'SOL',
+  explorerUrl: 'https://solscan.io',
+  rpcUrl: 'https://api.devnet.solana.com'
+}
+
+export interface WalletConnectAppMetadata {
+  name: string
+  description: string
+  url: string
+  icons: string[]
+}
+
+export class WalletConnectConnector extends BaseConnector implements Connector {
+  id = 'WalletConnect'
+  name = 'WalletConnect'
+  ready = true
+  protected provider: UniversalProvider | undefined
+  protected qrcode: boolean
+
+  public constructor({
+    relayerRegion,
+    metadata,
+    qrcode,
+    autoconnect
+  }: {
+    relayerRegion: string
+    metadata: WalletConnectAppMetadata
+    qrcode?: boolean
+    autoconnect?: boolean
+  }) {
+    super()
+    this.qrcode = Boolean(qrcode)
+    UniversalProviderFactory.setSettings({
+      projectId: SolStoreUtil.getProjectId(),
+      relayerRegion,
+      metadata,
+      qrcode: this.qrcode
+    })
+
+    UniversalProviderFactory.getProvider().then(provider => {
+      provider.on('session_delete', () => {
+        delete provider.session?.namespaces['solana']
+      })
+    })
+
+    if (autoconnect) {
+      UniversalProviderFactory.getProvider().then(provider => {
+        if (provider.session?.namespaces['solana']?.accounts?.length) {
+          const [defaultAccount] = provider.session.namespaces['solana'].accounts
+          const address = defaultAccount?.split(':')[2] ?? ''
+          SolStoreUtil.setIsConnected(true)
+          SolStoreUtil.setAddress(address)
+        }
+      })
+    }
+  }
+
+  public static readonly connectorName = 'walletconnect'
+
+  public async disconnect() {
+    const provider = await UniversalProviderFactory.getProvider()
+
+    try {
+      await provider.disconnect()
+    } finally {
+      // (TODO update typing for provider)
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+      delete provider.session?.namespaces['solana']
+    }
+
+    SolStoreUtil.setAddress('')
+  }
+
+  public override getConnectorName(): string {
+    return WalletConnectConnector.connectorName
+  }
+
+  public override async getProvider() {
+    const provider = await UniversalProviderFactory.getProvider()
+
+    return provider
+  }
+
+  public async signMessage(message: Uint8Array) {
+    const address = SolStoreUtil.state.address
+    if (!address) {
+      throw new Error('No signer connected')
+    }
+
+    const signedMessage = await this.request('solana_signMessage', {
+      message: base58.encode(message),
+      pubkey: address
+    })
+    const { signature } = signedMessage
+
+    return signature
+  }
+
+  public async signVersionedTransaction(transaction: VersionedTransaction) {
+    const transactionParams = {
+      feePayer: new PublicKey(SolStoreUtil.state.address ?? '').toBase58(),
```

Comment:
What happens when doing `PublicKey('')`? Will it throw / show weird behaviors?

Reply:
@valentyn-solana If this `''` check is in responsible of the end user, we should definitely disable them from running unexpected situations. There might be a check inside the function like the following:

```js
  public async signVersionedTransaction(transaction: VersionedTransaction) {
     if(!SolStoreUtil.state.address) { ... take any action or throw error } 
   ...
   }
```

This has better readability because the engineers know what happens if the address is empty.